gemelli/preprocessing.py

```diff
@@ -217,16 +216,19 @@ def _get_taxonomy_io_stream(taxonomy):
     highest_rank_indx = taxonomy.dropna(axis=0, how='any').first_valid_index()
     rank_order = [rank[0]
                   for rank in taxonomy.loc[highest_rank_indx].values.tolist()]
-    nl.set_rank_order(rank_order)
 
-    # collapse taxonomy back into a single string
-    taxonomy.fillna('', inplace=True)
+    fill_missing_dict = {i: f'{r}__' for i, r in enumerate(rank_order)}
+    # collapse taxonomy_ back into a single string
+    taxonomy.fillna(value=fill_missing_dict, inplace=True)
```

Comment:
Does pandas index into `fill_missing_dict` by column name? I'm not familiar with this use

Reply:
Should a unit test assert what's going on with these changes?